cmake/modules/SwiftSource.cmake

```diff
@@ -78,6 +78,17 @@ function(handle_swift_sources
       list(APPEND swift_compile_flags "-Xcc" "-D__SWIFT_CURRENT_DYLIB=swiftCore")
     endif()
 
+    precondition(NOT SWIFTSOURCES_IS_STDLIB OR NOT SWIFTSOURCES_MODULE_DIR,
+      "Cannot provide MODULE_DIR if IS_STDLIB is set")
+
+    if(SWIFTSOURCES_IS_STDLIB)
```

Comment:
@jrose-apple Would it be sufficient to change `if(SWIFTSOURCES_IS_STDLIB)` to `if(SWIFTSOURCES_TARGET_LIBRARY)` and pass the `TARGET_LIBRARY` param into `handle_swift_sources`?